cns/types/types.go

```diff
@@ -499,6 +499,62 @@ func init() {
 	types.Add("CnsQueryVolumeInfoResult", reflect.TypeOf((*CnsQueryVolumeInfoResult)(nil)).Elem())
 }
 
+type CnsRelocateVolumeRequestType struct {
+	This          types.ManagedObjectReference `xml:"_this"`
+	RelocateSpecs []BaseCnsVolumeRelocateSpec  `xml:"relocateSpecs"`
+}
+
+func init() {
+	types.Add("CnsRelocateVolumeRequestType", reflect.TypeOf((*CnsRelocateVolumeRequestType)(nil)).Elem())
+}
+
+type CnsRelocateVolume CnsRelocateVolumeRequestType
+
+func init() {
+	types.Add("CnsRelocateVolume", reflect.TypeOf((*CnsRelocateVolume)(nil)).Elem())
+}
+
+type CnsRelocateVolumeResponse struct {
+	Returnval types.ManagedObjectReference `xml:"returnval"`
+}
+
+type CnsVolumeRelocateSpec struct {
+	types.DynamicData
+
+	VolumeId  CnsVolumeId                           `xml:"volumeId"`
+	Datastore types.ManagedObjectReference          `xml:"datastore"`
+	Profile   []types.BaseVirtualMachineProfileSpec `xml:"profile,omitempty,typeattr"`
+}
+
+func init() {
+	types.Add("CnsVolumeRelocateSpec", reflect.TypeOf((*CnsVolumeRelocateSpec)(nil)).Elem())
+}
+
+type CnsBlockVolumeRelocateSpec struct {
+	XSIType string `default:"CnsBlockVolumeRelocateSpec" xml:"http://www.w3.org/2001/XMLSchema-instance type,attr,omitempty"`
```

Comment:
Don't need this field with the `typeattr` added to the `CnsRelocateVolumeRequestType.RelocateSpecs` field tag

Reply:
Thanks for the tip. :)

Reply:
made requested changes.